src/command.js

```diff
@@ -2,74 +2,115 @@ import program from 'commander';
 import fs from 'fs';
 import path from 'path';
 import YAML from 'yamljs';
-import { defaultBlockNumber } from './config';
-import { isAddress, validateBlockNumber } from './web3/utils';
+import { defaultBlockNumber, watchingConfigPath, getEnv, saveStateFileName } from './config';
+import { isAddress, getLastBlock } from './web3/utils';
 import { isPathExist } from './utils';
 
 /**
  * convert string to array
- * @param {string} val 
+ * @param {string} val
  * @return Array
  */
 const list = val => val.split(',');
 
 
-export default (filePath, lastBlockNumber) => {
-  const watchConfig = YAML.load(filePath);
-  program
-    .version('0.1.0')
-    .option('-a, --addresses <n>', 'List of address', list)
-    .option('-f, --from [n]', 'From block', typeof watchConfig.from !== 'undefined' ? watchConfig.from : defaultBlockNumber)
-    .option('-t, --to [n]', 'To block', typeof watchConfig.to !== 'undefined' ? watchConfig.to : defaultBlockNumber)
-    .option('-q, --quick [n]', 'Quick Mode', typeof watchConfig.quick !== 'undefined' ? watchConfig.quick : false)
-    .option('-s, --save-state [n]', 'Save state', typeof watchConfig.saveState !== 'undefined' ? watchConfig.saveState : null);
+export class Commands {
+  constructor(watchPath, lastBlockNumber) {
+    const watchConfig = watchPath ? YAML.load(watchPath) : YAML.load(watchingConfigPath);
+    program
+      .version('0.1.0')
+      .option('-a, --addresses [n]', 'List of address', list, Commands.handelInputValues('ADDRESSES', watchConfig.addresses, ''))
+      .option('-f, --from [n]', 'From block', Commands.handelInputValues('FROM_BLOCK', watchConfig.from, 0))
+      .option('-t, --to [n]', 'To block', Commands.handelInputValues('TO_BLOCK', watchConfig.to, defaultBlockNumber))
+      .option('-q, --quick [n]', 'Quick Mode', Commands.handelInputValues('QUICK_MODE', watchConfig.quick, true))
+      .option('-s, --save-state [n]', 'Save state', Commands.handelInputValues('SAVE_STATE', watchConfig.saveState, null))
+      .option('-n, --node-url [n]', 'Node address', Commands.handelInputValues('RPC_URL', watchConfig.node, 'http://localhost:8545'))
+      .option('-l, --log-level [n]', 'Log level', Commands.handelInputValues('LOG_LEVEL', watchConfig.log_level, 'info'))
+      .option('-p, --socket-port [n]', 'IO socket port number', Commands.handelInputValues('SOCKET_IO_PORT', watchConfig.socket, 3030))
+      .option('-o,--output-type [n]', 'Output type', Commands.handelInputValues('OUTPUT_TYPE', watchConfig.outputType, 'terminal'))
```

Comment:
'3030' shouldn't be hard coded here, it should be config variable.

Reply:
Do we use `SOCKET_IO`?

Reply:
Do we use socket io when dealing with graylog? i just used what i found in the env file if that's not the case then it should have been removed sometime ago